extension/background.js

```diff
@@ -1,79 +1,107 @@
+ * the chrome menu. Once created, it will set the image data for chrome.
+ * @param color A hexadecimal color formatted in hex.
+ * @param contributions The number of contributions for a given day.
+ */
 function draw(color, contributions) {
-  var canvas = document.createElement('canvas');
-  canvas.width = 19;
-  canvas.height = 19;
+    var canvas = document.createElement('canvas');
+    canvas.width = 19;
+    canvas.height = 19;
 
-  var context = canvas.getContext('2d');
-  context.fillStyle = color;
-  context.fillRect(0, 0, 19, 19);
+    var context = canvas.getContext('2d');
+    context.fillStyle = color;
+    context.fillRect(0, 0, 19, 19);
 
-  color_hex = parseInt("0x" + color.slice(1));
-  if (color_hex < 0x44a340) {
-      text_color = "#FFFFFF";
-  } else {
-      text_color = "#000000";
-  }
-  context.fillStyle = text_color;
-  context.textAlign = "center";
-  context.textBaseline = "middle";
-  context.font = "11px Arial";
-  context.fillText(contributions, 10, 10);
+    var color_hex = parseInt("0x" + color.slice(1));
+    // Ternary operator to determine the fill style.
+    context.fillStyle = color_hex < 0x44a340 ? "#FFFFFF" : "#000000";
+    context.textAlign = "center";
+    context.textBaseline = "middle";
+    context.font = "11px Arial";
+    context.fillText(contributions, 10, 10);
 
-  chrome.browserAction.setIcon({
-    imageData: context.getImageData(0, 0, 19, 19)
-  });
+    chrome.browserAction.setIcon({
+        imageData: context.getImageData(0, 0, 19, 19)
+    });
 }
 
+/**
+ * Handle the contributions data by parsing through the temporary dom
+ * Then calls draw to build the icon for the extension icon
+ * @param response_dom
+ */
 function handle_contrib_data(response_dom) {
     var weeks = response_dom.getElementsByTagName("g");
-    weeks = weeks[weeks.length-1];
+    weeks = weeks[weeks.length - 1];
     var days = weeks.getElementsByTagName("rect");
-    var last_day = days[days.length-1];
+    var last_day = days[days.length - 1];
 
     var todays_contributions = last_day.getAttribute("data-count");
     var color = last_day.getAttribute("fill");
     draw(color, todays_contributions);
 }
 
-
+/**
+ * Get the contributions for the given user and parse the data
+ */
 function get_contrib() {
-    console.log(username);
-    var contrib_url = "https://github.com/users/" + username + "/contributions";
-    var xhr = new XMLHttpRequest();
-    xhr.onreadystatechange = function(){
-        if (xhr.readyState == 4) {
+    request('get', "https://github.com/users/" + username + "/contributions")
+        .then(function (xhr) {
+            console.log(username);
+
             var parser = new DOMParser();
             var dom = parser.parseFromString(xhr.responseText, "text/html");
             handle_contrib_data(dom);
-        }
-    };
-    xhr.open("GET", contrib_url, true);
-    xhr.send();
+        });
 }
 
+/**
+ * Make AJax request and call closure when ready.
```

Comment:
Honestly, I as going for AJAX but looks like my shift key gave out. :/ Do you have any preferences?